pkg/addon/source.go

```diff
@@ -392,23 +150,63 @@ func NewAsyncReader(baseURL, bucket, subPath, token string, rdType ReaderType) (
 			bucketEndPoint = fmt.Sprintf(bucketTmpl, ossURL.Scheme, bucket, ossURL.Host)
 		}
 		return &ossReader{
-			baseReader:     bReader,
 			bucketEndPoint: bucketEndPoint,
 			path:           subPath,
 			client:         resty.New(),
 		}, nil
 	}
-	return nil, errors.New("addon registry invalid")
+	return nil, fmt.Errorf("invalid addon registry type '%s'", rdType)
+}
+
+// BuildReader will build a AsyncReader from registry, AsyncReader are needed to read addon files
+func (r Registry) BuildReader() (AsyncReader, error) {
+	if r.OSS != nil {
+		o := r.OSS
+		return NewAsyncReader(o.Endpoint, o.Bucket, o.Path, "", ossType)
+	} else if r.Git != nil {
```

Comment:
```suggestion
	}
	if r.Git != nil {
```

In fact, we don't need to use `else if` as it will return if above condition matches. We should make it as clean as possible.